codegen/src/main/twirl/templates/JavaClient/Client.scala.txt

```diff
@@ -139,8 +151,30 @@ public abstract class @{service.name}Client extends @{service.name}ClientPowerAp
       public java.util.concurrent.CompletionStage<akka.Done> closed() {
         return channel.closedCS();
       }
+
+      /**
+       * The same client instance decorated to add the given key and value to the metadata of any request issued.
+       */
+      public @{service.name}Client addRequestHeader(String key, String value) {
+        return new Default@{service.name}Client(
+          channel,
+          isChannelOwned,
+          settings,
+          options,
+          mat,
+          ec,
+          defaultMetadata.addEntry(key, value));
+      }
   }
 
+   /**
+    * The same client instance decorated to add the given key and value to the metadata of any request issued.
+    */
+   public @{service.name}Client addRequestHeader(String key, String value) {
+     // default implementation ignoring header for source compatibility
+     return this;
```

Comment:
I'm struggling to understand why.. given this does not exist currently, it will only be generated when people upgrade.. so then, what is the possible source of incompatibility if this actually adds an header instead?

Reply:
They could be extending the client interface with their own stub in tests, and then that would no longer compile because missing the method if we didn't have this default imp.

Reply:
I see, thanks.